src/BondEvaluatorQuartic.h

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2018-2020, Michael P. Howard
+// Copyright (c) 2021-2024, Auburn University
+// Part of azplugins, released under the BSD 3-Clause License.
+
+#ifndef AZPLUGINS_BOND_EVALUATOR_QUARTIC_H_
+#define AZPLUGINS_BOND_EVALUATOR_QUARTIC_H_
+
+#ifndef __HIPCC__
+#include <string>
+#endif
+
+#include "BondEvaluator.h"
+
+#ifdef __HIPCC__
+#define DEVICE __device__
+#else
+#define DEVICE
+#endif
+
+namespace hoomd
+    {
+namespace azplugins
+    {
+namespace detail
+    {
+
+//! Parameters of quartic bond potential
+struct BondParametersQuartic : public BondParameters
+    {
+#ifndef __HIPCC__
+    BondParametersQuartic()
+        : k(0), r_0(0), b_1(0), b_2(0), U_0(0), sigma_6(0), epsilon_x_4(0), delta(0)
+        {
+        }
+
+    BondParametersQuartic(pybind11::dict v)
+        {
+        k = v["k"].cast<Scalar>();
+        r_0 = v["r_0"].cast<Scalar>();
+        b_1 = v["b_1"].cast<Scalar>();
+        b_2 = v["b_2"].cast<Scalar>();
+        U_0 = v["U_0"].cast<Scalar>();
+        delta = v["delta"].cast<Scalar>();
+
+        auto sigma(v["sigma"].cast<Scalar>());
+        auto epsilon(v["epsilon"].cast<Scalar>());
+
+        const Scalar sigma_2 = sigma * sigma;
+        const Scalar sigma_4 = sigma_2 * sigma_2;
+        sigma_6 = sigma_2 * sigma_4;
+        epsilon_x_4 = Scalar(4.0) * epsilon;
+        }
+
+    pybind11::dict asDict()
+        {
+        pybind11::dict v;
+        v["k"] = k;
+        v["r_0"] = r_0;
+        v["b_1"] = b_1;
+        v["b_2"] = b_2;
+        v["U_0"] = U_0;
+        v["sigma"] = pow(sigma_6, 1. / 6.);
+        v["epsilon"] = epsilon_x_4 / Scalar(4.);
+        v["delta"] = delta;
+        return v;
+        }
+#endif
+
+    Scalar k;           //!< k parameter for quartic potential strength
+    Scalar r_0;         //!< energy barrier breaking distance
+    Scalar b_1;         //!< quartic tuning parameter #1
+    Scalar b_2;         //!< quartic tuning parameter #2
+    Scalar U_0;         //!< quartic energy barrier to "breaking"
+    Scalar sigma_6;     //!< sigma raised to the power of 6 for WCA
+    Scalar epsilon_x_4; //!< epsilon * 4 for WCA
+    Scalar delta;       //!< delta parameter bond
```

Comment:
Do you have a use case for this `delta` parameter? Core shifting is no longer part of the "standard" WCA in HOOMD, so we could remove it if not to simplify the code below. Let me know, as that will inform my other comments!

Reply:
Hi Michael, thank you for having a look! I just double checked, and yes we do have a use case for it. We use it for consistency, as we have some systems with shifted lennard-jones pair potentials alongside these quartic bonds as a sort of secondary pseudo-pair potential.